This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit e11d3e6b277a4163071f8772499c1e72e67bcd91
Author: zhangjingqiang <zhangjingqiang@bytedance.com>
Date:   Thu Apr 21 11:56:06 2022 +0800

    update to uuid 1.0

Index: libsystemd/src/id128.rs
===================================================================
--- libsystemd.orig/src/id128.rs
+++ libsystemd/src/id128.rs
@@ -67,7 +67,7 @@ impl Id128 {
 
     /// Return this ID as a lowercase hexadecimal string, with dashes.
     pub fn dashed_hex(&self) -> String {
-        format!("{}", self.uuid_v4.to_hyphenated_ref())
+        format!("{}", self.uuid_v4.hyphenated())
     }
 
     /// Custom serialization (lower hex).
Index: libsystemd/Cargo.toml
===================================================================
--- libsystemd.orig/Cargo.toml
+++ libsystemd/Cargo.toml
@@ -63,7 +63,7 @@ version = "^0.10"
 version = "^1.0"
 
 [dependencies.uuid]
-version = "^0.8.1"
+version = "^1.0"
 features = ["serde"]
 [dev-dependencies.pretty_assertions]
 version = "^1.0"
